### DISCUSSION TITLE
Is there any secrets to make this plugin work faster?

### DISCUSSION THREAD

**Discussion by setofaces** [UnitTestBot/UTBotJava#2229] [Q&A] [answered]
Hello there. Liberica JDK17, IntelliJ IDEA 2023.1, Windows 10,  generation of tests even for thin classes like simple Date conversion util takes my whole life, is it expected behaviour at this stage of the product or I probably have some wrong settings in my idea enviroment which intefere with plugin work? Thank you.

**Comment by tyuldashev** [comment, marked as answer]
@setofaces It should be limited by "**Test generation timeout**" (per class) specified in test generation dialog. If it takes significantly more than set there, then it's a problem, otherwise just adjust this value to desired value.

![image](https://github.com/UnitTestBot/UTBotJava/assets/10035839/46a0b925-04a6-409e-a866-1eb59049ccba)

**Reply by setofaces** [reply]
thanks, didnt think timeout affects overall time.
